### DISCUSSION TITLE
Migrate self-developed plugins from Kong 2.8 to 3.0

### DISCUSSION THREAD

**Discussion by mong0520** [Kong/kong#9735] [Help] [answered]
hi guys, 

I have 2 questions about migrating self-developed plugins from Kong 2.8 to 3.0.

1) I found that the plugin's `:new()` function was not called in 3.0. 
> Ref: https://docs.konghq.com/gateway/latest/plugin-development/custom-logic/
> _You don’t need to add a :new() method or call any of the CustomHandler.super.XXX:(self) methods._

What's the replacement for this?

for example:
```lua
function MyPlugin:new()
  -- some initialization process goes here
end

function MyPlugin:access(conf)
  -- my own logic
end
```

2) I found that I cannot read environment variables by `os.getenv("KONG_MY_ENV)` anymore. Is that an expected behavior? I originally can do that in Kong 2.8.3

**Comment by hbagdi** [comment, marked as answer]
Discussed in https://github.com/Kong/kong/issues/9736
